### ISSUE TITLE
Coverity sudden improvement

### ISSUE THREAD

**Issue by elmart** [neovim/neovim#2168]
This is just to report that we have gone through a sudden improvement in our coverity score.
Last time I checked it (about a week ago), we were at 0.43 defect density with about 100 issues yet to solve.
Now it says 0.28 defect density and just 58 outstanding issues. Checking the recently fixed ones, you can see that, indeed, there were 44 issues that were last seen at March 11th, and have disappeared since.
I can't explain this looking at commits since March 11th, so I suppose there has been an update on the server side which somehow has improved logic and eliminated previous false positives. I'll try to confirm with Coverity staff.
If that's confirmed, it would be great news: I repeat: 0.28 defect density and just 58 issues to go. We're nearer to a bug-free (as far as static analysis can go) neovim!

**Comment by justinmk** [comment]
3baba1e7bc6698e6bc9f1d37fce88b30d6274bc9 removed `term.c`, I wonder how many warnings were associated with that?

**Comment by elmart** [comment]
I don't think it's because of that. That commit was merged 23 days ago, and vanished issues were still there less that a week ago (on March 11th). Also, I've checked the fixed issues and they are scattered through many different files (curiously, none of them term.c).

**Comment by tarruda** [comment]
> If that's confirmed, it would be great news: I repeat: 0.28 defect density and just 58 issues to go. We're nearer to a bug-free (as far as static analysis can go) neovim!

Thats great news!

**Comment by Pyrohh** [comment]
> 33 new defect(s) introduced to neovim/neovim found with Coverity Scan.

:(

**Comment by elmart** [comment]
Yes. Unfortunately, it seems I was right about a server change having caused the sudden improvement. This is the mail I sent to Coverity staff:

```
Hi,
 
I'm elmart from neovim.
We have experienced a sudden improvement in our coverity score.
It was 0.43 around a week ago, with about 100 outstanding issues.
It is 0.28 right now, with 58 outstanding issues.
I've checked and it seems 44 issues were last seen by March 11th and have disappeared since.
I can't explain this through commits in our repository, so I've thought maybe a server side update has improved logic so that all those false positives have gone.
Could you confirm that, please?
This is, was there a server update shortly after March 11th?
 
Regards,
elmart
```

And this is what they have answered:

```
Hi,

It looks like we had some settings change on server side that caused this behavior.
We have reverted back and it shows the right number.

Thanks
Dakshesh Vyas • Technical Manager
dvyas@coverity.com • 185 Berry Street, Suite 6500, San Francisco CA 94017
Office: +1 (415) 935 2957 
scancoverity.com www.coverity.com • Twitter: @coverity @coverityScan
Coverity by Synopsys
```

Thus the 33 new issues.
I'm not sure it's completely fixed, though, as they were 44 issues that "disappeared" and only 33 to have been restablished. It's a pity, but Coverity seems a bit unstable sometimes.

**Comment by elmart** [comment]
I checked again, and though it's true only 33 issues are marked as "new", if you go to the "newly detected" tab, then you get the 44 issues that "disappeared" before. I think the difference is: new -> never seen before, and "newly detected" -> detected in current snapshot, and not detected in previous one (but possibly detected in past). But, then, I don't understand why 33 of them are marked as "new". As I said, Coverity is not always consistent regarding classification of issues.

In any case, I think current state is ok, and it's the best we're going to have. Closing this.
